### ISSUE TITLE
Is it possible to stub generators?

### ISSUE THREAD

**Issue by jpetitcolas** [cjohansen/Sinon.JS#612]
I'm looking for a way to stub a generator, and I can't find any solution. Here is one of my attempt:

``` js
var sinon = require('sinon');

var myObj = {
  myGenerator: function* () {
    yield "Hello world!";
  }
};

console.log(myObj.myGenerator().next().value); // Hello world!

myObj.myGenerator = sinon.stub().returns(function* () {
  return "I'm in stub!";
});

console.log(myObj.myGenerator().toString()); // function* () { yield "I'm a stub!"; }

console.log(myObj.myGenerator().next().value);
/**
 * Raises following error:
 * 
 * console.log(myObj.myGenerator().next().value);
 *                                 ^
 * TypeError: undefined is not a function
 */
```

Am I missing something? Or are generators not supported?

I found a [fork](https://github.com/ingameio/SinonES6.JS) which seems to support them, but I'd like to keep the official version for latest updates. :)

**Comment by limianwang** [comment]
I have the same trouble. I'm having problem wrapping my head around how to use sinon and ES6 features, especially generators.. Any one having the same issue?

**Comment by mroderick** [comment]
A fork is being suggested in https://github.com/cjohansen/Sinon.JS/issues/638, perhaps that can work for you.

Pull requests welcome!

**Comment by zhengyhn** [comment]
Is anyone working on this so far?

**Comment by fatso83** [comment]
@zhengyhn don't think so. Feel free to start a PR✌ I wrote a whole bunch of tips on what to think about in #638 you should read in that case.

**Comment by fatso83** [comment]
Thinking some more about this I think I practically solved the entire issue in #638, just waiting for someone to get it done, but I might as well finish it now seeing that no-one ever did. Give me some ⌚
